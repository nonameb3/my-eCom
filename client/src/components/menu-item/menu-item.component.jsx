import React from "react";
import { withRouter } from "react-router-dom";
import "./menu-item.style.scss";

function Menuitem(props) {
  const { title, size, imageUrl, history, match, linkUrl } = props;
  return (
    <div
      className={`${size} menu-item`}
      onClick={() => history.push(`${match.url}${linkUrl}`)}
    >
      <div
        className="background-image"
        style={{
          backgroundImage: `url(${imageUrl})`
        }}
      />
      <div className="content">
        <div className="title">{title}</div>
        <span className="subtitle">SHOP NOW</span>
      </div>
    </div>
  );
}

export default withRouter(Menuitem);
