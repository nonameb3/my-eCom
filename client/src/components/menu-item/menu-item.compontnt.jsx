import React from "react";
import "./menu-item.sttyle.scss";

export default function Menuitem({ title, size, imageUrl }) {
  return (
    <div className={`${size} menu-item`}>
      <div
        className="background-image"
        style={{
          backgroundImage: `url(${imageUrl})`
        }}
      />
      <div className="content">
        <div className="title">{title}</div>
        <span className="subtitle">SHOP NOW</span>
      </div>
    </div>
  );
}
