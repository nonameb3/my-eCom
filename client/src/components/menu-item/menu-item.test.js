import React from "react";
import { shallow } from "enzyme";

import { Menuitem } from "./menu-item.component";

let wrapper;
const mockHistory = {push: jest.fn()}
const mockProps = {
  title: "test-title",
  size: 99,
  imageUrl: "abc.com",
  history: mockHistory,
  match: {url:'/abc'},
  linkUrl: "/url.com"
};

describe('munu-item component', () => {
  beforeEach(() => {
    wrapper= shallow(<Menuitem {...mockProps}/>);
  });

  it('should render menu-item component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should rander title as expect', () => {
    expect(wrapper.find('.title').text()).toEqual(mockProps.title)
  });

  it('should action history.push when click', () => {
    wrapper.find('.menu-item').simulate('click');
    expect(mockHistory.push).toHaveBeenCalledWith(`${mockProps.match.url}${mockProps.linkUrl}`);
  });
})
