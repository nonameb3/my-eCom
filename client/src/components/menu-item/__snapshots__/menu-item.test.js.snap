// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`munu-item component should render menu-item component 1`] = `ShallowWrapper {}`;
